packages/@aws-cdk/aws-sagemaker/lib/model.ts

```diff
@@ -0,0 +1,405 @@
+
+  /**
+   * Returns the name of this model.
+   *
+   * @attribute
+   */
+  readonly modelName: string;
+
+  /**
+   * The IAM role associated with this Model.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Adds a statement to the IAM role assumed by the instance.
+   */
+  addToRolePolicy(statement: iam.PolicyStatement): void;
+}
+
+/**
+ * Represents a Model resource defined outside this stack.
+ */
+export interface ModelAttributes {
+  /**
+   * The name of this model.
+   */
+  readonly modelName: string;
+
+  /**
+   * The IAM execution role associated with this model.
+   *
+   * @default - When not provided, any role-related operations will no-op.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * The security groups for this model, if in a VPC.
+   *
+   * @default - When not provided, the connections to/from this model cannot be managed.
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+}
+
+abstract class ModelBase extends cdk.Resource implements IModel {
+  /**
+   * Returns the ARN of this model.
+   * @attribute
+   */
+  public abstract readonly modelArn: string;
+  /**
+   * Returns the name of the model.
+   * @attribute
+   */
+  public abstract readonly modelName: string;
+  /**
+   * Execution role for SageMaker Model
+   */
+  public abstract readonly role?: iam.IRole;
+  /**
+   * The principal this Model is running as
+   */
+  public abstract readonly grantPrincipal: iam.IPrincipal;
+  /**
+   * An accessor for the Connections object that will fail if this Model does not have a VPC
+   * configured.
+   */
+  public get connections(): ec2.Connections {
+    if (!this._connections) {
+      throw new Error('Cannot manage network access without configuring a VPC');
+    }
+    return this._connections;
+  }
+  /**
+   * The actual Connections object for this Model. This may be unset in the event that a VPC has not
+   * been configured.
+   * @internal
+   */
+  protected _connections: ec2.Connections | undefined;
+
+  /**
+   * Adds a statement to the IAM role assumed by the instance.
+   */
+  public addToRolePolicy(statement: iam.PolicyStatement) {
+    if (!this.role) {
+      return;
+    }
+
+    this.role.addToPolicy(statement);
+  }
+}
+
+/**
+ * Describes the container, as part of model definition.
+ */
+export interface ContainerDefinition {
+  /**
+   * The image used to start a container.
+   */
+  readonly image: ContainerImage;
+
+  /**
+   * A map of environment variables to pass into the container.
+   *
+   * @default - none
+   */
+  readonly environment?: {[key: string]: string};
+
+  /**
+   * Hostname of the container.
+   *
+   * @default - none
+   */
+  readonly containerHostname?: string;
+
+  /**
+   * S3 path to the model artifacts.
+   *
+   * @default - none
+   */
+  readonly modelData?: ModelData;
+}
+
+/**
+ * Construction properties for a SageMaker Model.
+ */
+export interface ModelProps {
+
+  /**
+   * The IAM role that the Amazon SageMaker service assumes.
```

Comment:
Is there any more information / docs that we can point to that show what permissions SageMaker requires for this role?

Reply:
[These are the best docs](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-createmodel-perms) I could find. If those look good, would you like me to link to those?

Reply:
Perfect